podcast/2.17.md

```diff
@@ -0,0 +1,487 @@
+
+**Beyang Liu:**
+
+Awesome. And the thing it was doing in Atom was that code navigation as well, or was that more on the syntax highlighting side?
+
+**Max Brunsfeld:**
+
+We were just using it for syntax highlighting and code folding. So, you could fold functions that weren't necessarily indented in the way that Atom had previously required them to be. And also that command I mentioned earlier, extend-selection, we made that a built-in feature of Atom so you could grow your selection up the syntax tree. It's funny, it's this unassuming feature, but it was this feature of the IDEs that I had used to use a long time ago that I had loved so much. You could put multiple cursors on different codes that you wanted to refactor and grow them all up the syntax tree so they would all be selecting the right thing and then do refactoring.
+
+**Beyang Liu:**
+
+That sounds really satisfying.
+
+**Max Brunsfeld:**
+
+Yeah. I love it. So it didn't have a huge bang but I just loved it.
+
+**Beyang Liu:**
+
+That's awesome. What about Tree-sitter as an open source project? Because now it's popping up in a lot of places. I think more and more people are using it inside other editors and building on top of it. I know GitHub continues to build on top of it for their code navigation. Have you kept up with the goings-on of the project and where it's being used?
+
+**Max Brunsfeld:**
+
+I do. Yeah. I try to. It's been taken up more and more. I hear about companies that are paying people to work on grammars and parsers and stuff, but a big one was Semgrep. They make tools for searching for certain patterns and especially anti-patterns in source code and correcting them. They switched to using Tree-sitter. And then one that made me really happy, even though I've moved on from using VIM, was that Neovim incorporated a built-in feature of Neovim that you could parse code with Tree-Sitter.
+
+And what really, I think was the final thing that helped more people find a use for, was the library that we built. That was when I eventually left the Atom team while still at GitHub and went to work on the code navigation thing full-time and we built this feature in Tree-sitter that was about searching for patterns in a syntax tree. And our goal was to make it so that you could have one engine that was like a pattern-matching engine in a syntax tree that would both let us do syntax highlighting. You'd be able to say there was like a language that was based on S-expressions like Lisp where you could look for a function containing a return type of [law???] and you could syntax highlight that. And then we can use the same engine for matching things like that for recording functions, calls, and definitions for code navigation and we just called it tree queries.
```

Comment:
This is the section, [law???] specifically, I'm having trouble understanding

Reply:
It does sound like he says law, which could making sense here in a hypothetical function that queries an AST.